### PR TITLE
[SofaNG] Setup StateContainer

### PR DESCRIPTION
**Pull request by fredroy**
#1527

Based on master

Deprecates partially _SofaBaseMechanics_

It contains components used as a "State" in mapping, forcefield, etc. 
In reality, it is merely a module for MechanicalObject for the moment, as MappedObject is not really used anywhere
In the future, it could contain TemperatureContainer, etc

~~namespaces are not changed, which is nice.~~

Collegial decision to rename it to **StateContainer**

EDIT:
[ci-depends-on https://github.com/sofa-framework/SofaPython3/pull/239]
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 1]
```diff
@@ -818,6 +818,10 @@ const std::map< std::string, CreatableMoved, std::less<> > movedComponents =
     { "RecordedCamera", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
     { "Visual3DText", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
     { "VisualTransform", CreatableMoved("v22.06", "SofaGeneralVisual", "Sofa.Component.Visual") },
+
+    // SofaBaseMechanics was deprecated in #XXXX
```
Comment (alxbilger) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp]: Missing PR number

**Review by alxbilger — COMMENTED** [review]
I am not a fan of Container for this module. I think State or StateContainer is more meaningful.

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by guparan** [comment]
Are you dead, dear CI?
[ci-build][force-full-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][force-full-build][with-all-tests]

**Comment by damienmarchal** [comment]
OMG I'm so happy we are having some appropriate names :)

**Review by alxbilger — APPROVED** [review]

**Pull request merged**